File: project-euler-100-java/src/test/java/com/github/adrianomoreira/projecteuler100/Problem004Test.java
package com.github.adrianomoreira.projecteuler100;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class Problem004Test {

    @DisplayName("largestPalindromeProduct(2) should return 9009.")
    @Test
    public void test() {
        Assertions.assertEquals(9009, Problem004.largestPalindromeProduct(2));
    }

    @DisplayName("largestPalindromeProduct(3) should return 906609.")
    @Test
    public void test2() {
        Assertions.assertEquals(906609, Problem004.largestPalindromeProduct(3));
    }


}

File: project-euler/euler_dart/test/problem_004_test.dart
import 'package:test/test.dart';
import 'package:euler_dart/problem_004.dart';

void main() {
  test("largestPalindromeProduct(2) should return 9009", () {
    expect(largestPalindromeProduct(2), 9009);
  });
  test("largestPalindromeProduct(2) should return 9009", () {
    expect(largestPalindromeProduct(3), 906609);
  });
}
